src/main/java/ai/shreds/adapter/AdapterCategoryController.java

```diff
@@ -36,39 +35,29 @@ public AdapterCategoryController(ApplicationCreateCategoryInputPort createCatego
 
     @PostMapping
     public SharedCreateCategoryResponse createCategory(@Valid @RequestBody SharedCreateCategoryRequest request) {
-        validateCategoryData(request.getName(), request.getDescription());
         try {
             log.info("Creating category with name: {}", request.getName());
-            return new SharedCreateCategoryResponse(
-                createCategoryService.createCategory(request.getName(), request.getDescription()).getId(),
-                request.getName(),
-                request.getDescription()
-            );
+            return createCategoryService.createCategory(request.getName(), request.getDescription());
```

Comment:
the method must return a SharedUpdateCategoryResponse instead of SharedCategoryDTO